FreeRTOS/FreeRTOS-Plus-TCP
# DNS local lookups using mDNS repaired

Description
-----------
Somewhere along the way, active lookups using mDNS got broken.

When a peer answers an mDNS a lookup, in its answer, it will **not** repeat the questions that were asked by the client.

Test Steps
-----------

I did 3 kinds of lookups, each of them in 4 different ways:
A normal DNS lookup
A mDNS lookup of `laptop-hp.local`
An LLMNR lookup of `laptop-hp`

I repeated each lookup by either asking for A or for AAAA records. Also I would either use IPv4 or IPv6 to contact the peer/server, setting [`xDNS_IP_Preference`](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/4364e78a4a6413d11012cc3994a5f5c07df4af78/source/include/FreeRTOS_DNS.h#L91).

Here are the results of of mDNS/LLMNR tests:

[DNS_tests.zip](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/files/14611445/DNS_tests.zip)

Without the reparation, mDNS would not succeed.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
